### ISSUE TITLE
Show the Lobby chat whenever there is no ProgramItem chat to show

### ISSUE THREAD

**Issue by bcpierce00** [clowdr-app/clowdr-web-app#185]
At the moment, there are still a few ways to get the interface into a state where it's showing the session chat in the right-hand sidebar.  This is confusing.  Much better would be simply to make the RHS always show either the current ProgramItem or, if there isn't one, then the Lobby chat.

**Comment by bcpierce00** [comment]
... or perhaps the "Room chat" for the workshop that they are in?  This needs a bit more thought.

A related bug: getting to (for example) PLMW in two different ways displays two different text channels:
  - enter through Live Events --> PLMW channel
  - enter through PaaG --> Lobby

**Comment by jon-bell** [comment]
@bcpierce00 @sweirich please decide what behavior you would like here by EOD.

**Comment by bcpierce00** [comment]
OK, here's what we think make sense:
  - When a programItem is displayed in the center pane, then either
       - that item's chat is displayed, if the Track has "text channels" checked, or
       - the (hopefully one and only) Room where that item is occurring should have its text chat displayed 
  - when no programItem is displayed, the RH sidebar should always display Lobby

This is nearly the way it works now, modulo the bug reported above.  Entering PLMW though PaaG should display the PLMW (room) chat, not Lobby.

We think this will work for ICFP.  More broadly, though, there is probably a conversation to be had here about items, tracks, rooms, sessions, oh my...

**Comment by jon-bell** [comment]
I think that I correctly implemented what we discussed on Zoom - please double check and if it looks OK you can close this.

**Comment by bcpierce00** [comment]
Seems to mostly work, but I did notice one little issue.  I clicked first on an ICFP item, which correctly brought up the paper chat.  Then I clicked on an item with no paper, and the chat did not change away from the paper...  (I didn't try refreshing, unfortunately, and the second time I tried the same experiment it worked as expected.)

![image](https://user-images.githubusercontent.com/7014411/90835554-f0649580-e31a-11ea-9990-390ed55270f2.png)

**Comment by jon-bell** [comment]
Maybe something got lost in the middle (#193) - when in a *room* that has a paper going *now* that has no paper chat, what should show? If not the room chat (what I had thought we had discussed today), then why are we even calling it a room chat, and why don’t we just have channels named after things that people want to see?

**Comment by bcpierce00** [comment]
I think the current behavior is going to give us what we want for the workshops, but I agree there is potential for confusion.

Fundamentally, as I've said elsewhere, I think the problem is that the collection of IRL concepts that we're inheriting from Researchr (and from our IRL assumptions and biases) is just not quite fitting the virtual space.  The whole bundle -- tracks, rooms, sessions, papers, presentations of papers, Q&A for papers, etc., etc., -- needs to be taken apart and put together differently.

Text chats are about quick communication with a group of people who are "doing the same thing" as I am -- either doing the same thing right now, synchronously, or maybe doing the same thing at different times.  

A big confounding factor is "subtyping" (which PL people know always messes things up!): because events are nested (this paper is part of this session is part of this workshop is part of this conference week), the question "What are you doing now?" often has no clear answer.  

Slack deals with this (for one level of nesting only) using threads.  But we all agree that this is clunky even compared to the rest of slack.

**Comment by jon-bell** [comment]
We can (in the database) make multiple rooms use the same chat channel. Does this help?

**Comment by sweirich** [comment]
Can we force some rooms to use the lobby as their chat?  I think this is what we want for the icfp program rooms. (We have paper chats too, so room chats are just clutter. )

--Stephanie

**Comment by jon-bell** [comment]
Yes, that is an elegant solution to the problem. So, we are agreed though that #193 is fix-in-docs (that is, the behavior observed in that case is correct?)
